catroid/src/main/java/org/catrobat/catroid/content/actions/SetThreadColorAction.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2022 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.content.actions
+
+import android.util.Log
+import com.badlogic.gdx.graphics.Color
+import com.badlogic.gdx.scenes.scene2d.actions.TemporalAction
+import org.catrobat.catroid.content.Scope
+import org.catrobat.catroid.content.Sprite
+import org.catrobat.catroid.formulaeditor.Formula
+import java.io.IOException
+
+open class SetThreadColorAction : TemporalAction() {
+    private var scope: Scope? = null
+    private var color: Formula? = null
+    private var sprite: Sprite? = null
+    override fun update(delta: Float) {
+        try {
+            var colorStringInterpretation = "#ff0000"
+            if (color != null) {
+                colorStringInterpretation = color?.interpretString(scope).toString()
+            }
+            val red = "0x" + colorStringInterpretation.substring(SUBSTRING_POS_FIRST, SUBSTRING_POS_THIRD)
+            val green = "0x" + colorStringInterpretation.substring(SUBSTRING_POS_THIRD, SUBSTRING_POS_FIFTH)
+            val blue = "0x" + colorStringInterpretation.substring(SUBSTRING_POS_FIFTH, SUBSTRING_POS_SEVENTH)
+            val redInt = Integer.decode(red)
+            val greenInt = Integer.decode(green)
+            val blueInt = Integer.decode(blue)
+            val colorInterpretation = Color()
+            val colorValue = argbToInt(redInt, greenInt, blueInt)
+            Color.argb8888ToColor(colorInterpretation, colorValue)
+
+            sprite?.embroideryThreadColor = colorInterpretation
+        } catch (exception: IOException) {
```

Comment:
Please change this so that it catches the exceptions that can occur in this code (`StringIndexOutOfBoundsException`, `NumberFormatException`). The handling stays the same: The occurence will be logged, but other than that the action should just fail and leave `embroideryThreadColor` unchanged.